papermill/api.py

```diff
@@ -221,24 +221,54 @@ def display_output(self, name):
         if name not in outputs:
             raise PapermillException(
                 "Output Name '%s' is not available in this notebook.")
-        output = outputs[name]
-        ip_display(output.data, metadata=output.metadata, raw=True)
+
+        # Loop over elements of output
+        for output in outputs[name]:
+            ip_display(output.data, metadata=output.metadata, raw=True)
 
 
 def _get_papermill_metadata(nb, name, default=None):
     """Returns a dictionary of a notebook's metadata."""
     return nb.metadata.get('papermill', {}).get(name, default)
 
 
+def _cell_has_output_tag(cell):
+
+    for output in cell.get('outputs', []):
+        if 'papermill' in output.get('metadata', {}) and output.metadata.papermill.get('name'):
+            return True
+
+    return False
+
+
+def _get_cell_output_tag(cell):
+
+    if _cell_has_output_tag(cell):
+
+        names = set([output.metadata.papermill.get('name') for output in cell.outputs if 'papermill' in output.get('metadata', [])])
+
+        # If there are conflicting names, throw a shoe
+        if len(names) != 1:
+            raise NotImplemented
+
+        return list(names)[0]
+
+    else:
+        return None
+
+
 def _get_notebook_outputs(nb_node):
     """Returns a dictionary of the notebook outputs."""
+
     outputs = {}
+
     for cell in nb_node.cells:
-        for output in cell.get('outputs', []):
-            if 'papermill' in output.get('metadata', {}):
-                output_name = output.metadata.papermill.get('name')
-                if output_name:
-                    outputs[output_name] = output
+
+        name = _get_cell_output_tag(cell)
+
+        if name is not None:
+            outputs[name] = cell.get('outputs', [])
```

Comment:
This should check for the types of cells and only do something if it is a code_cell: https://nbformat.readthedocs.io/en/latest/format_description.html#code-cells 

Then you can assume the existence of outputs as outputs are required for code cells and are not used in any other kind of cell.

https://github.com/jupyter/nbformat/blob/3af03b7804357d967aa035d677c5be64c8a86680/nbformat/v4/nbformat.v4.schema.json#L184
```
            "required": ["cell_type", "metadata", "source", "outputs", "execution_count"],
```